packages/fuels-programs/src/receipt_parser.rs

```diff
@@ -120,6 +134,26 @@ impl ReceiptParser {
         None
     }
 
+    #[cfg(not(feature = "legacy_encoding"))]
+    fn extract_return_data(&mut self) -> Option<Vec<u8>> {
```

Comment:
This is a bit confusing for anyone without proper context. Can you write some comments here explaining why we're doing this?

Reply:
@hal3e is working on PR to remove the legacy encoding. I duplicated some methods here and hid them behind the legacy/not legacy flag so that @hal3e can have an easier time refactoring.

Reply:
I've updated to hold two separate implementations of the `ReceiptParser`